### PR TITLE
ui: also send the state of the CPU when using toggle pause

### PR DESCRIPTION
**Pull request by sfluor**
Now when using the "toggle pause" button we also receive the color palette and the CPU state

### CODE REVIEW

File: debugger/debugger.go [hunk 1]
```diff
@@ -81,6 +82,7 @@ func (db *Debugger) createServer(addr string) {
 
 func (db *Debugger) pause(w http.ResponseWriter, r *http.Request) {
 	db.emu.TogglePause()
+	db.sendState(w)
```
Comment (remicalixte) [inline debugger/debugger.go]: This handles both pause and unpause, maybe only send the state when pausing. when unpausing we don't really know what we are sending

File: debugger/debugger.go [hunk 2]
```diff
@@ -113,3 +108,17 @@ func (db *Debugger) breakpoint(w http.ResponseWriter, r *http.Request) {
...
+
+	res["palette"] = db.emu.PPU.ExportPalette()
+	res["cpu"] = db.emu.CPU
+	jsonRes, err := json.Marshal(res)
+	if err != nil {
+		return errors.Wrap(err, "error marshalling the emulator state")
+	}
+
+	_, err = w.Write(jsonRes)
+	return errors.Wrap(err, "error writing response")
```
Comment (remicalixte) [inline debugger/debugger.go]: Maybe we can use the new `%w` fmt flag instead?

File: debugger/debugger.go (hunk 1 above)
Comment (sfluor) [reply debugger/debugger.go]: :+1:

File: core/core.go [hunk 3]
```diff
@@ -215,8 +207,32 @@ func (e *Emulator) TogglePause() {
...
+	}
+	{
+		f, err := os.Create("/tmp/vram.snes")
+		if err != nil {
+			log.Error("failed to dump color palette", zap.Error(err))
+		} else {
+			defer f.Close()
+			f.Write(e.PPU.vram.bytes[:])
+		}
+	}
```
[the remark this answers is not in the document]
Comment (sfluor) [reply core/core.go]: oops

**Pull request merged**